foxpuppet/windows.py

```diff
@@ -2,44 +2,86 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from contextlib import contextmanager
-
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-class Windows(object):
+
+
+class WindowManager(object):
 
     def __init__(self, selenium):
         self.selenium = selenium
-        self._handle = None
 
     @property
-    def all(self):
-        """Returns all current window handles"""
-        return self.selenium.window_handles
+    def windows(self):
+        """
+        Sets all current window handles to appropriate window instances
+        :returns: A list of Window Instances
+        """
+        return [BrowserWindow(self.selenium, handle=handle)
+                for handle in self.selenium.window_handles]
 
-    @property
-    def current(self):
-        """Returns current window handle"""
-        return self.selenium.current_window_handle
+    def create_window_instance(self, handle):
```

Comment:
Why do we need this? Can't we just create the window instance where it's needed? I can understand that we might need a window factory once we have more than one window type - is that what you're preparing for? If so, I would suggest a `WindowFactory` class with a `create` method. I'm fine with leaving this until we have another window type though.